src/main/java/com/codahale/metrics/newrelic/NewRelicReporterFactory.java

```diff
@@ -0,0 +1,61 @@
+package com.codahale.metrics.newrelic;
+
+import com.codahale.metrics.MetricAttribute;
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.ScheduledReporter;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import com.newrelic.telemetry.Attributes;
+import com.newrelic.telemetry.SimpleMetricBatchSender;
+import com.newrelic.telemetry.metrics.MetricBatchSenderBuilder;
+import io.dropwizard.metrics.BaseReporterFactory;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.util.EnumSet;
+import java.util.Map;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+
+@JsonTypeName("newrelic")
+public class NewRelicReporterFactory extends BaseReporterFactory {
+  @JsonProperty private String apiKey = null;
+
+  @JsonProperty private String overrideUri = null;
+
+  @JsonProperty private Map<String, Object> commonAttributes = null;
+
+  @Valid @NotNull @JsonProperty
+  private EnumSet<MetricAttribute> disabledMetricAttributes = EnumSet.noneOf(MetricAttribute.class);
+
+  @Override
+  @NotNull
+  public ScheduledReporter build(final MetricRegistry registry) {
+    final MetricBatchSenderBuilder metricBatchSender = SimpleMetricBatchSender.builder(apiKey);
+    if (overrideUri != null) {
+      final URI uri = URI.create(overrideUri);
+      try {
+        metricBatchSender.uriOverride(uri);
+      } catch (MalformedURLException t) {
+        throw new IllegalArgumentException(t.getMessage(), t);
+      }
+    }
+    final Attributes attributes = new Attributes();
+    if (commonAttributes != null) {
+      for (final Map.Entry<String, Object> entry : commonAttributes.entrySet()) {
```

Comment:
remove `final`

Reply:
Sorry I disagree here.

Reply:
here's one of very many references: http://archive.vn/ZIwdk

Reply:
see above comment. Let's stick with the style in place in the codebase, rather than trying to change the style in only one class.